openedx/core/djangoapps/user_api/models.py

```diff
@@ -42,15 +42,29 @@ def get_value(cls, user, preference_key):
         Arguments:
             user (User): The user whose preference should be set.
             preference_key (str): The key for the user preference.
+            default: The object to return if user does not have preference key set
 
         Returns:
-            The user preference value, or None if one is not set.
+            The user preference value, or default if one is not set.
         """
         try:
             user_preference = cls.objects.get(user=user, key=preference_key)
             return user_preference.value
         except cls.DoesNotExist:
-            return None
+            return default
+
+    @classmethod
+    def get_all_preferences(cls, user):
```

Comment:
Why introduce this method instead of calling `user_api.preferences.api.get_user_preferences`?